src/dotnet/commands/dotnet-install/dotnet-install-tool/InstallToolCommand.cs

```diff
@@ -67,69 +70,68 @@ internal class InstallToolCommand : CommandBase
 
             _shellShimMaker = shellShimMaker ?? new ShellShimMaker(executablePackagePath.Value);
 
-            _reporter = reporter ?? Reporter.Output;
+            _reporter = reporter;
         }
 
         public override int Execute()
         {
+            var reporter = (_reporter ?? Reporter.Output);
+            var errorReporter = (_reporter ?? Reporter.Error);
             if (!_global)
             {
                 throw new GracefulException(LocalizableStrings.InstallToolCommandOnlySupportGlobal);
             }
 
-            var toolConfigurationAndExecutablePath = ObtainPackage();
-
-            var commandName = toolConfigurationAndExecutablePath.Configuration.CommandName;
-            _shellShimMaker.EnsureCommandNameUniqueness(commandName);
-
-            _shellShimMaker.CreateShim(
-                toolConfigurationAndExecutablePath.Executable.Value,
-                commandName);
+            try
+            {
+                var toolConfigurationAndExecutablePath = ObtainPackage();
 
-            _environmentPathInstruction
-                .PrintAddPathInstructionIfPathDoesNotExist();
+                var commandName = toolConfigurationAndExecutablePath.Configuration.CommandName;
+                _shellShimMaker.EnsureCommandNameUniqueness(commandName);
 
-            _reporter.WriteLine(
-                string.Format(LocalizableStrings.InstallationSucceeded, commandName));
+                _shellShimMaker.CreateShim(
+                    toolConfigurationAndExecutablePath.Executable.Value,
+                    commandName);
 
-            return 0;
-        }
+                _environmentPathInstruction
+                    .PrintAddPathInstructionIfPathDoesNotExist();
 
-        private ToolConfigurationAndExecutablePath ObtainPackage()
-        {
-            try
-            {
-                FilePath? configFile = null;
-                if (_configFilePath != null)
-                {
-                    configFile = new FilePath(_configFilePath);
-                }
-
-                return _toolPackageObtainer.ObtainAndReturnExecutablePath(
-                    packageId: _packageId,
-                    packageVersion: _packageVersion,
-                    nugetconfig: configFile,
-                    targetframework: _framework,
-                    source: _source);
+                reporter.WriteLine(
+                    string.Format(LocalizableStrings.InstallationSucceeded, commandName));
             }
-
             catch (PackageObtainException ex)
             {
-                throw new GracefulException(
-                    message:
-                    string.Format(LocalizableStrings.InstallFailedNuget,
-                        ex.Message),
-                    innerException: ex);
+                errorReporter.WriteLine(ex.Message);
```

Comment:
Now, this class is directly handing low level output and format of the error message?

Reply:
Yes, it should.  Throwing `GracefulException` should only be reserved for command line syntax errors as it purposely shows the help from the tool; that indicates to users they just need to "type something different to solve the problem.  For failures installing the tool or in the tool itself (such as configuration), the error should be logged by the command.

I should make this `.Red()` though like the base command uses for errors.

Reply:
".Red()" -- I haven't dug about the color issue too. I remember there is a problem if the user set the command line profile differently (like change back ground to white). I don't know if we just pic a red and it will work for all profile.

I don't know if we have a abstraction already. What we really want is a presentation to call attention.